common/djangoapps/util/tests/test_course.py

```diff
@@ -1,36 +1,131 @@
-
-    def get_about_page_link(self):
-        """ create mock course and return the about page link."""
-        course_key = SlashSeparatedCourseKey('mitX', '101', 'test')
-        return get_lms_link_for_about_page(course_key)
+
+from django.conf import settings
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from util.course import get_link_for_about_page
+
+
+@ddt.ddt
+class TestCourseSharingLinks(ModuleStoreTestCase):
+    """
+    Tests for course sharing links.
+    """
+    def setUp(self):
+        super(TestCourseSharingLinks, self).setUp()
+
+        # create test mongo course
+        self.course = CourseFactory.create(
+            org='test_org',
+            number='test_number',
+            run='test_run',
+            default_store=ModuleStoreEnum.Type.split,
+            social_sharing_url='test_social_sharing_url',
+        )
+
+        # load this course into course overview and set it's marketing url
+        self.course_overview = CourseOverview.get_from_id(self.course.id)
+        self.course_overview.marketing_url = 'test_marketing_url'
+        self.course_overview.save()
+
+    def get_course_sharing_link(self, enable_social_sharing, enable_mktg_site, use_overview=True):
+        """
+        Get course sharing link.
+
+        Arguments:
+            course: This can be either a course overview or a course descriptor
+            enable_social_sharing(Boolean): To indicate whether social sharing is enabled.
+            enable_mktg_site(Boolean): A feature flag to decide activation of marketing site.
+
+        Keyword Arguments:
+            use_overview: indicates whether course overview or course descriptor should get
+            past to get_link_for_about_page.
+
+        Returns course sharing url.
+        """
+        mock_settings = {
+            'FEATURES': {
+                'ENABLE_MKTG_SITE': enable_mktg_site
+            },
+            'SOCIAL_SHARING_SETTINGS': {
+                'CUSTOM_COURSE_URLS': enable_social_sharing
+            },
+        }
+
+        with mock.patch.multiple('django.conf.settings', **mock_settings):
+            course_sharing_link = get_link_for_about_page(
+                self.course_overview if use_overview else self.course
+            )
+
+        return course_sharing_link
+
+    @ddt.data(
+        (True, True, 'test_social_sharing_url'),
+        (False, True, 'test_marketing_url'),
+        (True, False, 'test_social_sharing_url'),
+        (False, False, '{}/courses/course-v1:test_org+test_number+test_run/about'.format(settings.LMS_ROOT_URL)),
+    )
+    @ddt.unpack
+    def test_get_correct_course_about_link(self, enable_social_sharing,
+                                           enable_mktg_site, expected_course_sharing_link):
```

Comment:
Maybe we can rename it to `test_course_about_link` to fit this on 1 line OR define the Arguments on new line.e.g.
```
test_get_correct_course_about_link(
self, enable_social_sharing,enable_mktg_site, expected_course_sharing_link
):
```
You can also set each arg on it's individual line.